### PR TITLE
Upgrade to elm/http 2.0.0

### PR DESCRIPTION
**Pull request by dvekeman**
Hi 

Here is an attempt to upgrade elm/httpbuilder to the latest elm/http version. I tried to minimize the impact on the api.

I wasn't able to test it on a real application (as I'm still trying to upgrade it to 0.19 :)) but I did try it out in the HTTP example from the elm documentation and I added it in an examples project.

I'm not sure if I got it right on the implementation of `toRequest` though

```
toRequest : RequestBuilder msg -> Cmd msg
toRequest builder =
    let req = if builder.withCredentials
              then Http.riskyRequest
              else Http.request

    in req
      -- ...
```

Any feedback welcome!

Regards
Dieter

### CODE REVIEW

**Comment by Natim** [comment]
This is huge, thanks for the effort. I am going to try it.

File: README.md [hunk 1]
```diff
@@ -37,7 +35,23 @@ addItem item =
...
+update msg model =
+    case msg of
+        GotItem (Ok items) ->
+          ( { model | items = items }
+          , Cmd.none
+          )
+
+        GotItem (Err err) ->
+          -- Handle the error...
+          let _ = Debug.log (Debug.toString err)
```
Comment (koenusz) [inline README.md]: Isn't this debug statement going to prevent this package from being compiled with --optimize?

Comment (dvekeman) [reply README.md]: Hi @koenusz 

Thanks for reviewing.
I updated the README and the example.

Regards
Dieter

**Comment by kornicameister** [comment]
@lukewestby any news ?

**Comment by lukewestby** [comment]
I'm gonna review this and either merge it or suggest changes on Monday

**Comment by lukewestby** [comment]
@dvekeman this is very good, thank you!

**Comment by dvekeman** [comment]
Awesome. Thanks a lot.

**Pull request merged**